src/Thumbnail.vue

```diff
@@ -0,0 +1,130 @@
+<template>
+  <div :class="['thumb-wrapper', addClass, {'thumb-circle': circle}]" :style="[getBorder, getFontSize, getFontColor, getBgColor]">
+    <img ref="pic" @load.once="computeImgSize" :src="src" class="thumb-image" :width="computedWidth" :height="computedHeight" :alt="alt" v-if="isImage">
+    <slot></slot>
+  </div>
+</template>
+
+<script>
+  const DEFAULT_SIZE = 100;
+
+  export default {
+    props: {
+      circle: {
+        type: Boolean,
+        default: false
+      },
+      addClass: {
+        type: String,
+        default: ''
+      },
+      alt: {
+        type: String,
+        default: ''
+      },
+      background: {
+        type: String,
+        default: ''
+      },
+      border: {
+        type: String,
+        default: ''
+      },
+      fontcolor: {
+        type: String,
+        default: ''
+      },
+      fontsize: {
+        type: String,
+        default: ''
+      },
+      size: {
+        type: String,
+        default: ''
+      },
+      src: {
+        type: String
+      },
+      text: {
+        type: String,
+        default: ''
+      },
+    },
+    computed: {
+      getBorder() {
+        // default is no border
+        return this.border === '' ? {} : {border: this.border};
+      },
+      getFontSize() {
+        // default is half of {size}
+        let fs = this.size/2;
+        fs = this.fontsize === '' ? fs : this.fontsize;
+        return {'font-size': fs + 'px'};
+      },
+      getFontColor() {
+        // default is no setting (inherit)
+        return this.fontcolor === '' ? {} : {'color': this.fontcolor};
+      },
+      getBgColor() {
+        // default is no setting (inherit)
+        return this.background === '' ? {} : {'background': this.background};
+      },
+      isImage() {
```

Comment:
The current use of `isImage()` is to control the conditional rendering of the `<img>` tag.  

If the author inputs an invalid URL, nothing will show up on the page. I feel we should always render the image regardless of whether the URL is valid or not. Since the author can receive immediate feedback from the browser if he provided an invalid `src`, instead of just a blank div.

Also, we might want to do [validation of URLs](https://stackoverflow.com/questions/13820477/input-tag-validation-for-url) using either a pattern or regex. Then show a warning either on console or text below the component. Since this affects `<pic>` as well, we should do that in another issue + PR if needed.

Reply:
Another nit. The `isImage()` does not describe what this function is doing. `hasSrc()` would be a better fitting name.

Reply:
> If the author inputs an invalid URL, nothing will show up on the page. I feel we should always render the image regardless of whether the URL is valid or not. Since the author can receive immediate feedback from the browser if he provided an invalid `src`, instead of just a blank div.

What do you mean by always rendering the image regardless of whether the URL is valid or not. Currently, as long as an `src` is specified (and is not an empty string), the component will add it to the `img` tag.

> Another nit. The `isImage()` does not describe what this function is doing. `hasSrc()` would be a better fitting name.

Ok

Reply:
I wrote this before you clarified that the `<thumbnail>` feature does not always require `src`.

No changes will be required for this.